salt/modules/win_mslicense.py

```diff
@@ -0,0 +1,580 @@
+"""
+Module for managing OS and software licenses on Windows,
+utilizing PowerShell and CIM methods.
+
+Invoke-CimMethod is supported in PowerShell 3.0, so
+the module can potentially work starting from Windows 7.
+
+Guaranteed tests were conducted on Windows 10.
+"""
+
+import logging
+import salt.utils.platform
+
+log = logging.getLogger(__name__)
+__virtualname__ = "mslicense"
+
+
+def __virtual__():
+    """
+    Only works on Windows
+    """
+    if salt.utils.platform.is_windows():
+        return __virtualname__
+    return (False, "Only Windows OS supported")
+
+
+def install(key: str):
+    """
+    Install the given product key.
+
+    https://learn.microsoft.com/en-us/previous-versions/windows/desktop/sppwmi/installproductkey-softwarelicensingservice
+
+    Returns:
+        tuple: tuple as operation status and payload
+
+    .. code-block:: cfg
+
+        (False, "error message")
+        (True,)
+
+    CLI Example:
+
+    .. code-block:: powershell
+
+        salt '*' mslicense.install XXXXX-XXXXX-XXXXX-XXXXX-XXXXX
+    """
+
+    cmd = "Invoke-CimMethod -Query 'Select * From SoftwareLicensingService' -MethodName InstallProductKey -Arguments @{{ProductKey='{0}'}}".format(
```

Comment:
I recommend adding escaping and/or validation to any arguments that are extrapolated directly as a string to a powershell command. The single quotes `'` you have help with most of it so I think the only character you have to worry about/deal with is a single quote; so, any single quotes need to be escaped which in powershell is with double single quotes `''`. So `key.replace("'", "''")`.